mongodb/docs-worker-pool
# [DOP-4033]: Replicate shared.mk steps within Autobuilder code

## Ticket

[DOP-4033](https://jira.mongodb.org/browse/DOP-4033)

## Notes

This story converts the contents of the `shared.mk` file into TypeScript code. 

## Testing

1. Checkout the branch and run `npm ci` as there are some new dependencies
2. Run `docker buildx build --file Dockerfile.local . --build-arg NPM_BASE_64_AUTH=<NPM_BASE_64_AUTH> --build-arg NPM_EMAIL=<NPM_EMAIL> -t autobuilder/local:latest `; the email and auth values can be found in parameter store
3. Update the default profile in `~/.aws/credentials` with temporary credentials from AWS; the credentials will have a randomly named profile that you should rename to default when pasting into the file
4. Run `docker run  -v ~/.aws:/home/docsworker/.aws -e ENHANCED=true -e MONGO_ATLAS_USERNAME="<ATLAS_USERNAME>" -e MONGO_ATLAS_PASSWORD="<ATLAS_PASSWORD>"  -e MONGO_ATLAS_HOST="<ATLAS_HOST>" autobuilder/local:latest` using the staging environment values
